File: app/src/main/java/com/permissionnanny/data/OngoingRequestDB.java
package com.permissionnanny.data;

import android.support.v4.util.ArrayMap;
import com.permissionnanny.lib.request.RequestParams;

/**
 */
public class OngoingRequestDB {

    private final NannyDB mDB;

    public OngoingRequestDB(NannyDB db) {
        mDB = db;
    }

    public void open() {
        mDB.open();
    }

    public void putOngoingRequest(String clientId, RequestParams request) {
        mDB.put(clientId, request);
    }

    public ArrayMap<String, RequestParams> getOngoingRequests() {
        return mDB.findVal(null, RequestParams.class);
    }

    public void delOngoingRequest(String clientId) {
        mDB.del(clientId);
    }
}

File: app/src/main/kotlin/com/permissionnanny/data/OngoingRequestDB.kt
package com.permissionnanny.data

import android.support.v4.util.ArrayMap
import com.permissionnanny.lib.request.RequestParams

/**
 */
class OngoingRequestDB(private val db: NannyDB) {

    fun open() {
        db.open()
    }

    fun putOngoingRequest(clientId: String, request: RequestParams) {
        db.put(clientId, request)
    }

    val ongoingRequests: ArrayMap<String, RequestParams>
        get() = db.findVal(null, RequestParams::class.java)

    fun delOngoingRequest(clientId: String) {
        db.del(clientId)
    }
}
